pxd.c

```diff
@@ -1601,15 +1669,29 @@ ssize_t pxd_remove(struct fuse_conn *fc, struct pxd_remove_out *remove)
 #endif
 		}
 
-		pxd_free_disk(pxd_dev);
-		device_unregister(&pxd_dev->dev);
+	spin_unlock(&pxd_dev->lock);
 
-		module_put(THIS_MODULE);
-	}
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,15,0)
+	pxd_free_disk(pxd_dev);
```

Comment:
some part of this code conflicts with the recent issue we found. cant do device_add/device_remove under spinlock.

Reply:
done